lib/tasks/ee.rake

```diff
@@ -0,0 +1,263 @@
+require 'yaml'
+require 'erb'
+
+class Database
+  RECOGNIZED_OPTIONS = %w(user host port socket)
+
+  def initialize(options = {})
```

Comment:
I opted not to do this in the end, since the way I invoke this is with just the key/value pairs from the configuration yaml file:

```ruby
    task :create do |t, args|
      db = Database.new(database_config)
      db.create
    end
```

Instead, I chose to raise an exception when the database option is unavailable:

```ruby
  def database
    @options['database'] || raise("Missing value for `database` in options")
  end
```